src/mds/MDSRank.cc

```diff
@@ -3847,3 +3854,27 @@ void MDSRank::send_task_status() {
 
   schedule_update_timer_task();
 }
+
+void MDSRank::schedule_inmemory_logger() {
+  timer.add_event_after(g_conf().get_val<double>("mds_extraordinary_events_dump_interval"),
+                        new LambdaContext([this]() {
+                          inmemory_logger();
+                        }));
+}
+
+void MDSRank::inmemory_logger() {
+  if (g_conf().get_val<double>("mds_extraordinary_events_dump_interval") &&
+      (client_eviction_dump || beacon.missed_beacon_ack_dump ||
```

Comment:
IMO the `g_conf().get_val<double>()` is a little expensive in the `tick()` thread. Because the `timer` need to take the global `mds_lock`.

Reply:
In that case, I need to bring back the `handle_conf_change` function. :(